lib/mayaUsd/ufe/UsdTransform3dRead.h

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/ufe/UsdSceneItem.h>
+
+#include <pxr/usd/usd/prim.h>
+
+#include <ufe/transform3d.h>
+#include <ufe/transform3dHandler.h>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! \brief Read-only implementation for USD object 3D transform information.
+//
+// Note that all calls to specify time use the default time, but this
+// could be changed to use the current time, using getTime(path()).
+
+class MAYAUSD_CORE_PUBLIC UsdTransform3dRead : public Ufe::Transform3dRead
```

Comment:
Is there any way we can convert this into a template, so that UsdTransform3dBase could publicly derive from
UsdTransform3dReadT<Ufe::Transform3d>, and share some code ?  You can then simply typedef UsdTransform3dReadT<Ufe::Transform3dRead> UsdTransform3dRead;
I realize that UsdTransform3dBase can't simply derive from UsdTransform3dRead, because it needs the full interface of Ufe::Transform3d, but using a template to remove code duplication might be nice.

Reply:
I will instead write a non-template implementation class and use it in both? That will avoid having code in the header.

Reply:
Up to you whichever is cleanest.  You can also do as per
https://git.autodesk.com/media-and-entertainment/ufe/blob/master/include/trie.imp.h
and put the template implementation into a separate file, only to be included by those files than instantiate the template.